corehq/apps/hqwebapp/models.py

```diff
@@ -63,22 +64,19 @@ def __repr__(self):
 
     def save(self, *args, **kwargs):
         MaintenanceAlert.get_active_alerts.clear(MaintenanceAlert)
+        MaintenanceAlert.objects.filter(end_time__lte=datetime.utcnow()).update(active=False)
```

Comment:
What's this about?  With the query filters below, it shouldn't be necessary to update `active` for the query to return the correct results.

Reply:
That makes sense if you're thinking of a unified concept of "active", but I don't think it's needed from a query optimization standpoint, and it'd be preferable to avoid surprising side-effects like this when possible.

Reply:
Do you think there's any value in leaving a "deactivate alert" button alive on the admin page until manually deactivated, then? Currently I have them being displayed as "expired" with no action available if the end time is in the past, so an alert could potentially forever be `active=True` in the database, though it would no longer retrieved by the `get_active_alerts` query.

Reply:
> Do you think there's any value in leaving a "deactivate alert" button alive on the admin page until manually deactivated, then?
I don't think so, because toggling that won't actually change anything, so it might be confusing.  Showing them as "expired" as you have now seems right.